app/controllers/async_info_controller.rb

```diff
@@ -24,6 +25,24 @@ def base_data
     end
   end
 
+  def access_token
+    info = Struct.new({ name: params[:name], email: params[:email], image: params[:image] })
```

Comment:
All of these calls to `Struct.new` will create new anonymous class objects on each invocation:

```
[4] forem(main)> Struct.new(:name)
#<Class:0x0000560318735930> < Struct
[5] forem(main)> Struct.new(:name)
#<Class:0x0000560318785520> < Struct
```

To avoid this you have two options:

1. Assign the generated class to a constant (outside this method):
    ```ruby
    Info = Struct.new(:name, :email, :image)

    def access_token
      Info.new(name: params[:name], ...)
      ...
    end
    ```
2. Create a named struct in the `Struct` namespace (again outside this method):
    ```ruby
     Struct.new("Info", :name, :email, :image)
     Struct::Info.new(...)
    ```

Reply:
Thanks for these suggestions @citizen428! I was playing around here with the Structs and had to modify this section quite a bit so Husky hooks would pass and it got messed up a bit, but going forward I think I'm going with option 1 here. This is what it looks like locally for me right now:

<img width="433" alt="Screen Shot 2021-09-10 at 08 33 25" src="https://user-images.githubusercontent.com/6045239/132870695-edb8e012-2d4b-48e0-bdf8-7c2cf55b65c8.png">

I'll push some more changes soon. I'm trying out a few different combinations here and trying to get to the "least hacky" solution possible 😄